src/Tools/dotnet-counters/CounterMonitor.cs

```diff
@@ -238,15 +236,15 @@ public async Task<ReturnCode> Monitor(
                     {
                         //Cancellation token should automatically stop the session
 
-                        _stdOutput.WriteLine($"Complete");
+                        _console.Out.WriteLine($"Complete");
                         return ReturnCode.Ok;
                     }
                 }
             }
-            catch (CommandLineErrorException e)
+            catch (DiagnosticToolException dte)
             {
-                _stdError.WriteLine(e.Message);
-                return ReturnCode.ArgumentError;
+                _console.Error.WriteLine(dte.Message);
+                return (ReturnCode)dte.ReturnCode;
```

Comment:
Nit: Cast is not needed, ReturnCode is already a ReturnCode.

Reply:
Thanks for the catch, was remnant from when it was an int